RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc

```diff
@@ -166,42 +167,45 @@ namespace {
 	if (ret==-1.f) return ret;
       }
 
+
       // parametrized dz and dr cut by using their error
       if (dz_par1[2]<std::numeric_limits<float>::max() || dr_par1[2]<std::numeric_limits<float>::max()) {
+
 	float maxDz_par1[3];
 	float maxDr_par1[3];
-	dzCut_par1(trk,nLayers,dz_par1, maxDz_par1);
-	drCut_par1(trk,nLayers,dr_par1, maxDr_par1);
-
-        Point bestVertex = getBestVertex(trk,vertices,minNVtxTrk); // min number of tracks 3
-	if (bestVertex.z() < -99998.) {
-	  bestVertex = beamSpot.position();
+	dzCut_par1(trk,nLayers,dz_par1,dz_exp, maxDz_par1);
+	drCut_par1(trk,nLayers,dr_par1,dr_exp, maxDr_par1);
+
+	float maxDz_par[3];
+	float maxDr_par[3];
+	std::copy(std::begin(maxDz_par1),std::end(maxDz_par1),std::begin(maxDz_par));
+	std::copy(std::begin(maxDr_par1),std::end(maxDr_par1),std::begin(maxDr_par));
+
+	// parametrized dz and dr cut by using d0 and z0 resolution
+	if (dz_par2[2]<std::numeric_limits<float>::max() || dr_par2[2]<std::numeric_limits<float>::max()) {      
```

Comment:
ciao
yes, it is intentional
they should always go together,
I though in this way the code would be faster and clear
anyhow, those cuts are used only @HLT 

hope it is fine
thanks